jobmon runtime error: slice bounds out of range

I try to move my backup cron to jobmon, but it raises an error.
Now it wrotes me that a job with the same id is allready running.

command:
`jobmon -id backup -re=false -s none -- pg_dumpall > ~/postgres/backups/$(date +%F).bak`

output:
```
panic: runtime error: slice bounds out of range

goroutine 17 [running]:
github.com/cloudradar-monitoring/cagent/pkg/jobmon.(*captureWriter).Write(0xc0001b4090, 0xc0001d8000, 0x1000, 0x8000, 0x1000, 0x0, 0x0)
        /go/src/github.com/cloudradar-monitoring/cagent/pkg/jobmon/writer.go:25 +0x2aa
io.copyBuffer(0x7e63a0, 0xc0001b4090, 0x7e6800, 0xc0001a2108, 0xc0001d8000, 0x8000, 0x8000, 0x0, 0x0, 0x1000000000001)
        /usr/local/go/src/io/io.go:404 +0x1fb
io.Copy(...)
        /usr/local/go/src/io/io.go:364
os/exec.(*Cmd).writerDescriptor.func1(0xc00001c360, 0x0)
        /usr/local/go/src/os/exec/exec.go:288 +0x63
os/exec.(*Cmd).Start.func1(0xc0001ae000, 0xc00019c2c0)
        /usr/local/go/src/os/exec/exec.go:409 +0x27
created by os/exec.(*Cmd).Start
        /usr/local/go/src/os/exec/exec.go:408 +0x58f
```

--- a/pkg/jobmon/writer.go
+++ b/pkg/jobmon/writer.go
@@ -22,7 +22,7 @@ func (w *captureWriter) String() string {
 func (w *captureWriter) Write(p []byte) (n int, err error) {
 	gotLen := len(p)
 	if gotLen >= w.n {
-		w.buf = p[gotLen-w.n-1:]
+		w.buf = p[gotLen-w.n:]
 	} else if gotLen > 0 {
 		newLength := len(w.buf) + gotLen
 		if newLength <= w.n {

--- a/pkg/jobmon/writer_test.go
+++ b/pkg/jobmon/writer_test.go
@@ -1,0 +1,57 @@
+package jobmon
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCaptureWriter(t *testing.T) {
+	t.Run("simple-test", func(t *testing.T) {
+		incomingStream := bytes.NewBufferString("")
+		cw := newCaptureWriter(incomingStream, 100)
+
+		_, _ = cw.Write([]byte(""))
+		assert.Equal(t, "", cw.String())
+
+		_, _ = cw.Write([]byte("0123456789"))
+		assert.Equal(t, "0123456789", cw.String())
+
+		_, _ = cw.Write([]byte(strings.Repeat("0123456789", 9)))
+		assert.Equal(t, strings.Repeat("0123456789", 10), cw.String())
+	})
+
+	t.Run("full-copy", func(t *testing.T) {
+		incomingStream := bytes.NewBufferString("")
+		cw := newCaptureWriter(incomingStream, 100)
+
+		_, _ = cw.Write([]byte(strings.Repeat("0123456789", 10)))
+		assert.Equal(t, strings.Repeat("0123456789", 10), cw.String())
+	})
+
+	t.Run("overflow", func(t *testing.T) {
+		incomingStream := bytes.NewBufferString("")
+		cw := newCaptureWriter(incomingStream, 10)
+
+		_, _ = cw.Write([]byte(strings.Repeat("0123456789", 2)))
+		assert.Equal(t, strings.Repeat("0123456789", 1), cw.String())
+	})
+
+	t.Run("one-byte-overflow", func(t *testing.T) {
+		incomingStream := bytes.NewBufferString("")
+		cw := newCaptureWriter(incomingStream, 10)
+
+		_, _ = cw.Write([]byte("01234567890"))
+		assert.Equal(t, "1234567890", cw.String())
+	})
+
+	t.Run("overflow-many", func(t *testing.T) {
+		incomingStream := bytes.NewBufferString("")
+		cw := newCaptureWriter(incomingStream, 100)
+
+		_, _ = cw.Write([]byte(strings.Repeat("0123456789", 20)))
+		assert.Equal(t, strings.Repeat("0123456789", 10), cw.String())
+	})
+}

